GUI/src/main/kotlin/ui/components/general/FileSelector.kt

```diff
@@ -0,0 +1,31 @@
+package ui.components.general
+
+import javax.swing.JFileChooser
+
+/**
+ * Opens a file chooser dialog and returns the selected file path.
+ *
+ * @param onResult The callback to be called when a file was selected.
+ * @return The selected file path, or null if no file was selected.
+ */
+fun openFileChooserAndGetPath(onResult: (String) -> Unit) {
+    val fileChooser = JFileChooser()
+    val result = fileChooser.showOpenDialog(null)
+    if (JFileChooser.APPROVE_OPTION == result) {
+        onResult(fileChooser.selectedFile.absolutePath)
+    }
+}
+
+/**
+ * Opens a file chooser dialog and returns the selected file path.
+ *
+ * @param onResult The callback to be called when a file was selected.
+ * @return The selected file path, or null if no file was selected.
+ */
+fun openSaveChooserAndGetPath(onResult: (String) -> Unit) {
```

Comment:
I suggest calling this openFileSaverAndGetPath(). Seems more consistent with the above Function.